s4e-backend/src/main/java/pl/cyfronet/s4e/geoserver/op/GeoServerOperations.java

```diff
@@ -123,12 +129,48 @@ public void createExternalShpDataStore(String workspace, String dataStoreName, S
         return list("coverageStore", url);
     }
 
-    public void createS3CoverageStore(String workspace, String coverageStore, String s3url) {
-        URI url = UriComponentsBuilder.fromHttpUrl(
-                geoServerProperties.getBaseUrl() + "/workspaces/{workspace}/coveragestores")
-                .buildAndExpand(workspace).toUri();
-        val entity = httpEntity(new CreateS3CoverageStoreRequest(workspace, coverageStore, s3url));
-        restTemplate().postForObject(url, entity, String.class);
+    public void createS3CoverageStore(String workspace, String coverageStore) {
+        InputStream file = getArchive(coverageStore);
+
+        Unirest.put(geoServerProperties.getBaseUrl() + "/workspaces/{ws}/coveragestores/{cs}/file.imagemosaic?configure=none")
+                .header("Content-Type", "application/zip")
+                .routeParam("ws", workspace)
+                .routeParam("cs", coverageStore)
+                .basicAuth(geoServerProperties.getUsername(), geoServerProperties.getPassword())
+                .connectTimeout(geoServerProperties.getTimeoutConnect().intValue() * 1000)
+                .socketTimeout(geoServerProperties.getTimeoutRead().intValue() * 1000)
+                .field("file", file)
+                .asEmpty();
+    }
+
+    public InputStream getArchive(String coverageStore) {
+        ClassLoader classloader = Thread.currentThread().getContextClassLoader();
+        String mosaic = "Name=" + coverageStore + "\n" +
+                "TypeName=scene_" + coverageStore + "\n" +
+                "Levels=5000,5000\n" +
+                "LevelsNum=1\n";
+        InputStream streamMosaic = new SequenceInputStream(new ByteArrayInputStream(mosaic.getBytes()), classloader.getResourceAsStream("geoserver/mosaic.properties"));
+        InputStream streamStore = classloader.getResourceAsStream("geoserver/datastore.properties");
+        ByteArrayOutputStream bos = new ByteArrayOutputStream();
+        try {
+            ZipOutputStream zos = new ZipOutputStream(bos);
```

Comment:
You should use try-with-resources here for all the streams apart from bos, otherwise there may be memory leaks in case of errors.